src/engine/store/engineStore.ts

```diff
@@ -24,6 +25,7 @@ export const createEngineStore = () =>
           updateSketchParams: createUpdateSketchParams(set),
           setSketchModuleItem: createSetSketchModuleItem(set),
           updateNodeValue: createUpdateNodeValue(set),
+          updateSelectedNode: createUpdateSelectedNode(set),
```

Comment:
I'd rather this functionality live inside of `appStore` instead of `engineStore`. The `engineStore` should purely be about what's going on inside the scene, nothing UI related. Think about a web project that's running a Hedron scene. It's going to care about nodes and inputs, etc, but it doesn't care about what the user has selected. By keeping it all separate I'm hoping `hedronEngine` will stay quite slim as a package for web projects. 

If we were following the naming convention for `appStore`, it would be something like:

- `activeNodeId`
- `setActiveNodeId`

However just to complicate things further, i'm wondering if we want this to be an extra level deep. Something like `sketchActiveNodeIds`, which would be an object of sketch IDs mapped to their current `activeNodeId` (or `null`). So when switching between sketches we don't lose the current open node for that sketch.

Reply:
good calls, I'll work your feedback in and ping ya when it's ready! thanks!

Reply:
Okay, think it's good now, was a pretty easy change to make and seems to work well for me.

Do we want to have the param deselect if you click it a second time? I'm leaning towards no, as you might be tweaking the value manually, would become a pain to wait and see if you are interacting with a sub-element of it, especially if that's something plugins can have control over. Thinking we just slap a close button on the param editing popup

On to getting familiar with storybook and adding something to make use of this selected state, I'll do that on a separate branch